Sources/SwiftInspectorVisitors/Tests/GenericRequirementVisitorSpec.swift

```diff
@@ -71,6 +71,44 @@ final class GenericRequirementVisitorSpec: QuickSpec {
         }
       }
 
+      context("visiting a syntax tree involving a protocol that has a left-side fully qualified generic constraint") {
+        it("finds the generic requirements") {
+          let content = """
+            public protocol SomeGenericProtocol: GenericProtocol where
+              FooModule.LeftType == RightType
```

Comment:
Would this code be value? I thought that the left type needed to be on the protocol itself? Seeing an example that compiles would help me.

Reply:
yeah I thought that too! Then I put the following code into a playground and it compiled:
```
extension Array where
  AnyObject == Element
{}
```

Not that I think we should write code like this, but we should probably support someone doing so.

Reply:
And since that example used extensions rather than protocols, check out this code which also compiles:
```
protocol GenericTest {
  associatedtype SomeType
}

protocol GenericWhereTest: GenericTest where AnyObject == SomeType {}
```

Reply:
Good to know.